core/src/main/scala/com/salesforce/op/stages/impl/feature/SmartTextVectorizer.scala

```diff
@@ -84,12 +91,29 @@ class SmartTextVectorizer[T <: Text](uid: String = UID[SmartTextVectorizer[T]])(
     val minLenStdDev = $(minLengthStdDev)
     val shouldCleanText = $(cleanText)
 
-    implicit val testStatsMonoid: Semigroup[TextStats] = TextStats.monoid(maxCard)
-    val valueStats: Dataset[Array[TextStats]] = dataset.map(_.map(computeTextStats(_, shouldCleanText)).toArray)
-    val aggregatedStats: Array[TextStats] = valueStats.reduce(_ + _)
+    implicit val textStatsMonoid: Semigroup[TextStats] = TextStats.monoid(maxCard)
+    implicit val nameDetectStatsMonoid: Semigroup[NameDetectStats] = NameDetectStats.monoid
+    implicit val statsTupleMonoid: Semigroup[StatsTuple] = new Tuple2Semigroup[TextStats, NameDetectStats]()
+
+    val (aggTextStats, aggNameDetectStats) = if (getSensitiveFeatureMode == Off) {
+      (
+        dataset.map(_.map(computeTextStats(_, shouldCleanText)).toArray).reduce(_ + _),
+        Array.fill[NameDetectStats](inN.length)(NameDetectStats.empty)
+      )
+    } else {
```

Comment:
Can you make the formatting of both if branches look the same since they're doing nearly the same thing?